src/langs/vala/Compiler.js

```diff
@@ -1,49 +1,79 @@
 import Gio from "gi://Gio";
 import dbus_previewer from "../../Previewer/DBusPreviewer.js";
-import { decode } from "../../util.js";
+import { createTmpDir, copy } from "../../util.js";
+import { setupValaProject } from "./vala.js";
+
+let ready_to_build = false;
```

Comment:
I did this to know if the Vala builddir was already setup (as it needs a meson file and the workbench API). This could be done checking if a build file (like `workbench.vala`) exists, but I went for this instead as it is one less I/O operation.